pandas/core/reshape/merge.py

```diff
@@ -744,6 +747,57 @@ def get_result(self) -> DataFrame:
 
         return result.__finalize__(self, method="merge")
 
+    def _anti_join_update(self):
```

Comment:
ok with this being a method on _MergeOperation but it should have NO side effects, e.g. simply return left, right and assign at this level.

Reply:
not sure if I understand what you mean by, being a method on _MergeOperation.

Reply:
this needs to execute in the `get_result()` portion, NOT during validation. validation shouldn't dot he actual computation.

IOW during _get_join_info could dispatch based on `how`.

Reply:
the actual calculation is still happening in `get_result()`. 
I have just changed the configuration of `left`, `right` and `how` to utilise the already existing joining methods. E.g. For `anti_left` I have changed the `left`, `right` and `how` to `left` such that `left join` would give the result

Reply:
hi @jreback - wondering if you got a chance to look at this? and if this implementation is fine?

Reply:
no the implementation is quite convoluted 

i am not sure how to fix what u have here